### PR TITLE
Fix: make terratag action work

### PR DESCRIPTION
**Pull request by eladmosh**
### Issue & Steps to Reproduce / Feature Request
terratag actions doesn't really work, it marked as passed but there is an error in its logs:
Screen Shot 2022-01-02 at 14 44 36

### Solution
because we use typescript we should change our imports from `import  core from '@actions/core';` to `import * as core from '@actions/core';`

### CODE REVIEW

**Comment by roni-frantchi** [comment]
@eladmosh looks good - can we please add something that will prevent this from happening again please?

**Comment by eladmosh** [comment]
> @eladmosh looks good - can we please add something that will prevent this from happening again please?

the main problem and the cause that we didn't notice that `terratag-action` did not work is because that in our `catch` block the call to `core.setFailed` failed because of the way we imported `@actions/core`.
After this fix it wont happen again and the step will be marked as failed.

**Comment by roni-frantchi** [comment]
@eladmosh if we were to run `tsc` would that assert that?  
Otherwise - maybe we need an "integration test"?  
In other words - you could add a GitHub action to this repository that will be run on every push (like cI) - that will attempt to run the latest dist as action on some template - essentially what manual tests you did on another repo - but automated on this repo.

File: lib/terratag-action.spec.ts [hunk 1]
```diff
@@ -44,7 +48,7 @@ describe('terratag action', () => {
     });
     it('should download terratag from expected url', () => {
       expect(mockedTC.downloadTool.mock.calls).toEqual([
-        ['https://github.com/env0/terratag/releases/download/v1.2.3/terratag_1.2.3_linux_amd64.tar.gz']
+        [`https://github.com/env0/terratag/releases/download/v1.2.3/terratag_1.2.3_${osPlatform}_amd64.tar.gz`]
```
Comment (roni-frantchi) [inline lib/terratag-action.spec.ts]: The cpu architecture may also be different:
```suggestion
        [`https://github.com/env0/terratag/releases/download/v1.2.3/terratag_1.2.3_${osPlatform}_${os.os.arch()}.tar.gz`]
```

File: lib/terratag-action.spec.ts [hunk 2]
```diff
@@ -72,7 +76,7 @@ describe('terratag action', () => {
     });
     it('should download terratag from expected url', () => {
       expect(mockedTC.downloadTool.mock.calls).toEqual([
-        ['https://github.com/env0/terratag/releases/download/v5.6.7/terratag_5.6.7_linux_amd64.tar.gz']
+        [`https://github.com/env0/terratag/releases/download/v5.6.7/terratag_5.6.7_${osPlatform}_amd64.tar.gz`]
```
Comment (roni-frantchi) [inline lib/terratag-action.spec.ts]:
```suggestion
        [`https://github.com/env0/terratag/releases/download/v5.6.7/terratag_5.6.7_${osPlatform}_${os.arch()}.tar.gz`]
```

File: test-action-template/main.tf [hunk 3]
```diff
@@ -0,0 +1,8 @@
+provider "aws" {
+  version = "~> 2.0"
+  region  = "us-east-1"
+}
+
+terraform {
+  required_version = ">= 0.12"
+}
```
Comment (roni-frantchi) [inline test-action-template/main.tf]:
```suggestion
}

```

File: .github/workflows/test-terratag-action.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+            cd test-action-template && terraform init
+      - uses: ./
+        env: 
+            ACTIONS_STEP_DEBUG: ${{ secrets.ACTIONS_STEP_DEBUG }}
+        with:
+          tags: |
+            {
+              "env0_environment_id":"40907eff-cf7c-419a-8694-e1c6bf1d1168","env0_project_id":"43fd4ff1-8d37-4d9d-ac97-295bd850bf94"
+            }
+          dir: test-action-template
```
Comment (roni-frantchi) [inline .github/workflows/test-terratag-action.yml]: Let's move these templates to a more common pattern like `tests/fixtures`
```suggestion
          dir: tests/fixtures

```

**Review by roni-frantchi — APPROVED** [review]
Nicely done @eladmosh  
Left you some minor comments to address you may merge when done

File: lib/terratag-action.spec.ts (hunk 1 above)
Comment (eladmosh) [reply lib/terratag-action.spec.ts]: ab7fa460886ed211c5db8cc2ae4bbcf4bceecfde
had to use `mapOS` and `mapArch` for some reason all our terratag tgz files have `amd64` as arch

File: lib/terratag-action.spec.ts (hunk 2 above)
Comment (eladmosh) [reply lib/terratag-action.spec.ts]: ab7fa46

File: .github/workflows/test-terratag-action.yml (hunk 4 above)
Comment (eladmosh) [reply .github/workflows/test-terratag-action.yml]: 266d567

File: test-action-template/main.tf (hunk 3 above)
Comment (eladmosh) [reply test-action-template/main.tf]: ?

**Pull request merged**